gcc/rust/checks/errors/borrowck/rust-bir-builder-expr-stmt.cc

```diff
@@ -433,9 +454,6 @@ ExprStmtBuilder::visit (HIR::LoopExpr &expr)
 {
   auto loop = setup_loop (expr);
 
-  push_goto (loop.continue_bb);
-
-  ctx.current_bb = loop.continue_bb;
   (void) visit_expr (*expr.get_loop_block ());
```

Comment:
We should probably replace this cast with a proper std call.
```suggestion
  std::ignore = visit_expr (*expr.get_loop_block ());
```

Reply:
@P-E-P  You sure? According to cppreference this usage is UB. 

> While the behavior of std::ignore outside of std::tie is not formally specified, some code guides recommend using std::ignore to avoid warnings from unused return values of [[nodiscard]] functions. 

https://en.cppreference.com/w/cpp/utility/tuple/ignore

Reply:
We've discussed this privately, here are the emerging points:
- This is recommended by [cpp core guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es48-avoid-casts)
- Many codebase uses it (@jdupak mentioned chromium)
- Current state of `std::ignore` [here](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2023/p2968r0.html)
- [Standard on `std::ignore`](https://eel.is/c++draft/tuple#creation)

@CohenArthur @tschwinge Any insight on this ?

Reply:
I'm in favor of using `std::ignore` here :)